File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while(left <= right) {
            int mid = left + (right - left) / 2;
            int curr = nums[mid];
            if (curr == target) {
                return mid;
            }
            if (curr < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return -1;
    }
}

File: 704-binary-search/704-binary-search.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        return searchRecursive(nums, target, 0, nums.size - 1)
    }
    fun searchRecursive(nums: IntArray, target: Int, left: Int, right: Int): Int {
        if (left > right) {
            return -1
        }
        val mid = left + ((right - left) / 2)
        if (nums[mid] == target) {
            return mid
        } else if (target < nums[mid]) {
            return searchRecursive(nums, target, left, mid - 1)
        } else {
            return searchRecursive(nums, target, mid + 1, right)
        }
    }
}